File: answers/C/@mykeels/15-compound-interest-calculator.c
#include <stdio.h>

int main() {
    double principal, rate;
    int timeInYears;

    printf("Enter the Principal: ");
    scanf("%lf", &principal);

    printf("Enter the rate (%%): ");
    scanf("%lf", &rate);

    printf("Enter the time in years: ");
    scanf("%d", &timeInYears);


    for (int year = 1; year <= timeInYears; year++) {
        double interest = (principal * rate / 100);
        printf("\nYear %d\n", year);
        printf("Compound Interest: %f\n", interest);
        principal = principal + interest;
        printf("New Principal: %f\n", principal);
    }

    return 0;
}

File: answers/Go/@codehakase/15-compound-interest-calculator.go
package main

import "fmt"

func main() {
	var principal, rate float64
	var timeInYears int

	fmt.Println("Enter the principal: ")
	fmt.Scanf("%f\n", &principal)

	fmt.Println("Enter the rate %: ")
	fmt.Scanf("%f\n", &rate)

	fmt.Println("Enter the time in years: ")
	fmt.Scanf("%d\n", &timeInYears)
	fmt.Println()

	for year := 1; year <= timeInYears; year++ {
		interest := (principal * rate / 100)
		fmt.Printf("Year %d\n", year)
		fmt.Printf("Compound Interest: %.2f\n", interest)
		principal += interest
		fmt.Printf("New principal: %.2f\n", principal)
	}
}
